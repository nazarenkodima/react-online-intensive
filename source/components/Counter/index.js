// Core
import React from 'react';
import propTypes from 'prop-types';

//Instrumens
import Styles from './styles.m.css';

const Counter = ({count}) => (
    <section className = { Styles.counter }>Post count: { count }</section>
);

Counter.propTypes = {
    count: propTypes.number,
};

Counter.defaultProps = {
    count: 0,
};

export default Counter;
