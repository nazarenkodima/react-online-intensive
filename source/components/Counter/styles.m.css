.counter {
    display: flex;
    justify-content: center;
    position: relative;
    color: var(--paletteColor7);

    &::after {
        content: '';
        position: absolute;
        top: -6px;
        left: -35px;
        width: 30px;
        height: 30px;
    }
}
