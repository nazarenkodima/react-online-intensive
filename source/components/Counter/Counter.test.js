// Core
import React from 'react';
import Counter from './';
import renderer from 'react-test-renderer';

const renderTree = renderer.create(<Counter count = { 3 }/>).toJSON();

test('counter component should correspond to its snapshot counterpart', () => {
    expect(renderTree).toMatchSnapshot();
});

