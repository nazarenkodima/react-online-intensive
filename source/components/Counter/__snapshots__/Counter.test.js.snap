// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter component should correspond to its snapshot counterpart 1`] = `
<section
  className="counter"
>
  Post count: 
  3
</section>
`;
